Android/app/src/main/java/com/example/boardgamesocial/API/RetrofitClient.java

```diff
@@ -73,8 +63,9 @@ public static RetrofitClient getClient() {
         return new Gson().fromJson(jsonObject, cls);
     }
 
-    public void setAuthToken(Token token) {
-        HeaderInterceptor.token.setToken(token.getToken());
+    //Changed from Token Object to String token as SharedPreference is used for user session
+    public void setAuthToken(String token) {
+        HeaderInterceptor.token.setToken(token);
```

Comment:
Can we use Token as an object in SharedPreferences? If we use String then that will probably change authentication flow and test BeforeClass method.

Reply:
Okay, shall find a way around for that

Reply:
We can do that: [https://stackoverflow.com/questions/7145606/how-do-you-save-store-objects-in-sharedpreferences-on-android](url)

Reply:
Actually should be relatively just need to change:
This:
```java
@POST("login/")
    Call<Token> loginCall(
            @Body User user
    );
```  
To this:
```java
@POST("login/")
    Call<JsonObject> loginCall(
            @Body User user
    );
```
And store token(JsonObject) that you received in `onResponse` for login call in pref. Then once we need it we can use `getObject` from `RetrofitClient` class to get token object that can be set with `setAuthToken`

Reply:
Actually disregard this. I will update tests and call on the login page to match string token setter, it will be more consistent this way